upack/test/src/upack/BoolTests.scala

```diff
@@ -0,0 +1,32 @@
+package upack
+
+import utest._
+
+object BoolTests extends TestSuite {
+  def tests = Tests{
+    test("upack.Bool apply") {
+      upack.Bool(true) ==> upack.True
+      upack.Bool(false) ==> upack.False
+    }
+
+    test("upack.Bool.value") {
+      upack.Bool(true).value ==> true
+      upack.Bool(false).value ==> false
+    }
+    test("upack.Bool unapply") {
+      for(bool <- Seq(true, false)){
+        val jsb = upack.Bool(bool)
+        assertMatch(jsb) {
+          case upack.Bool(value)
+            if value == bool
+            && jsb.value == value =>
+        }
+      }
+    }
+    test("upack.Bool pattern matching should be exhaustive") {
+      upack.Bool(true) match {
+        case upack.Bool(value) =>
+      }
```

Comment:
This test assumes we're using `-Xfatal-warnings`, right? If not, I don't see how this test is supposed to detect non-exhaustiveness.

Reply:
Yes, exactly. I added it in the previous PR exactly for this test. I don't know if there is a better way to test this.

Reply:
That means, if the settings get changed later, we have no indicator that this test is insufficient. The bare minimum is a comment in code or in the test name. A bit better would be to detect the flags (e.g. by generating some build info code). I know utest has `compileError`; maybe it also fails a warning under `-Xfatal-warnings`, in which case we could guard some code that emits a warning with `compileError` to detect when this setting gets removed in the future.